### PR TITLE
Add new IDF conversion configs

### LINKED ISSUE #36: Add new IDF conversion configuration files
The following IDF configuration files must be integrated:

cmems_001_024_hourly_mean_surface.idf
cmems_001_024_hourly_smoc.idf
cmems_008_046.idf
cmems_015_003_0m.idf
cmems_015_003_15m.idf
esa_cci_sst.idf
ghrsst_l2p_modis_a_jpl_day.idf
ghrsst_l2p_modis_a_jpl_night.idf
ghrsst_l2p_viirs_jpl.idf
ghrsst_l2p_viirs_navo.idf
ghrsst_l2p_viirs_ospo.idf
sentinel1_l2_rvl.idf

### PR DESCRIPTION
**Pull request by aperrin66**
Resolves #36
I had to refactor the IDFConverter classes to manage the multiple conversion cases.
Sorry for the big PR, I had to do it all at once to make sure that all configuration cases were taken into account.

### CODE REVIEW

File: geospaas_processing/downloaders.py [hunk 1]
```diff
@@ -96,9 +96,8 @@ def check_and_download_url(cls, url, download_dir, **kwargs):
         if not file_name:
             raise DownloadError(f"Could not find file name for '{url}'")
         file_path = os.path.join(download_dir, file_name)
-        LOGGER.info("file_path %s", file_path)
```
Comment (aanersc) [inline geospaas_processing/downloaders.py:99]: is that unnecessary that you remove it?

File: geospaas_processing/downloaders.py [hunk 2]
```diff
@@ -96,9 +96,8 @@ def check_and_download_url(cls, url, download_dir, **kwargs):
...
+
         if os.path.exists(file_path) and os.path.isfile(file_path):
-            LOGGER.info("file exists")
```
Comment (aanersc) [inline geospaas_processing/downloaders.py:101]: is that unnecessary that you remove it?

Comment (aperrin66) [reply geospaas_processing/downloaders.py:101]: that was a message I used for debugging when I was coding this, and I forgot to remove it

File: geospaas_processing/downloaders.py (hunk 1 above)
Comment (aperrin66) [reply geospaas_processing/downloaders.py:99]: that was a message I used for debugging when I was coding this, and I forgot to remove it

**Comment by aperrin66** [comment]
I noticed a bug to fix on the conversion of Sentinel 1 RVL files, I will mark the PR as ready to review when it is fixed.

**Review by akorosov — DISMISSED** [review]

**Pull request merged**